main.py

```diff
@@ -5031,6 +5026,102 @@ def do_command(self, command, args, sender, channel, tags, isWhisper=False):
                     self.message(channel, "... and this was enough to upgrade your hand to a new slot! naroYay",
                                  isWhisper)
                 return
+            if command == "godimage":
+                canManageImages = sender in superadmins
+                godRarity = int(config["numNormalRarities"]) - 1
+                if len(args) < 1:
+                    if canManageImages:
+                        self.message(channel, "Usage: !godimage change / changeglobal / queue / check / acceptsingle / acceptglobal / reject", isWhisper)
+                    else:
+                        self.message(channel, "Usage: !godimage change / changeglobal / list / cancel", isWhisper)
+                    return
+                subcmd = args[0].lower()
+                if subcmd in ["change", "changeglobal", "request", "requestglobal"]:
+                    do_global = subcmd in ["changeglobal", "requestglobal"]
+                    if len(args) < 3:
+                        self.message(channel, "Usage: !godimage change[global] <id> <link>", isWhisper)
+                        return
+                    try:
+                        waifuid = int(args[1])
+                    except ValueError:
+                        self.message(channel, "Usage: !godimage change[global] <id> <link>", isWhisper)
+                        return
+
+                    with db.cursor() as cur:
+                        cur.execute("SELECT COUNT(*) FROM has_waifu WHERE userid = %s AND waifuid = %s AND rarity = %s", [tags['user-id'], waifuid, godRarity])
```

Comment:
I think this assumes that there's only one God card in a player's hand. Given this might not be necessarily the case (there's the potential special case of a duplicated God via Miku's Mercy), you may want to error out if the count is > 1.

Reply:
Way too much of this code has to rely on gods being singular for it to function sanely. I think Miku's Mercy will have to not apply to Gods, regardless of Maren's generosity.

Reply:
Not apply to Gods where it would cause a second copy to exist, that is.

Reply:
(Basically, I'm not rewriting the whole thing just to cover one extremely rare and extremely dumb edge case)